### PR TITLE
Add Sales Orders to Project costing and billing section 

### PR DESCRIPTION
**Pull request by superlack**
To provide the same level of "dashboard" financial tracking for projects to work alongside external clients that use PO's

### CODE REVIEW

File: erpnext/selling/doctype/sales_order/sales_order.py [hunk 1]
```diff
@@ -167,10 +167,20 @@ def on_cancel(self):
...
 		self.update_reserved_qty()
-
+		self.update_project()
 		self.update_prevdoc_status('cancel')
 
 		frappe.db.set(self, 'status', 'Cancelled')
+		
+	def update_project(self):
+		project_list = []
+		for d in self.items:
```
Comment (nabinhait) [inline erpnext/selling/doctype/sales_order/sales_order.py]: "Project" field does not exists in Sales Order Item table, it is in the parent table.

**Review by nabinhait — CHANGES_REQUESTED** [review]
Travis failing

**Comment by superlack** [comment]
Will fix, originally tested with "total" fields in SO.  Thanks

**Comment by superlack** [comment]
I fixed the errors (to the best of my knowledge), but Travis is showing errors on the following
`test_overlap_with_half_day (erpnext.hr.doctype.leave_application.test_leave_application.TestLeaveApplication) ... ERROR
test_overlap_with_half_day_not_applicable (erpnext.hr.doctype.leave_application.test_leave_application.TestLeaveApplication) ... ERROR`

File: erpnext/projects/doctype/project/project.py [hunk 2]
```diff
@@ -176,6 +176,13 @@ def update_purchase_costing(self):
 			from `tabPurchase Invoice Item` where project = %s and docstatus=1""", self.name)
 
 		self.total_purchase_cost = total_purchase_cost and total_purchase_cost[0][0] or 0
+		
+	def update_sales_costing(self):
+		total_sales_cost = frappe.db.sql("""select sum(total)
```
Comment (nabinhait) [inline erpnext/projects/doctype/project/project.py]: Should not it be `grand_total`?

Comment (superlack) [inline erpnext/projects/doctype/project/project.py]: You're right, I was thinking about how the billing hours don't show taxes, but those are further malleable unlike SO/SINV

**Comment by superlack** [comment]
I hope that's everything

**Review by nabinhait — APPROVED** [review]

**Pull request merged**